workflow/snakemake_rules/main_workflow.smk

```diff
@@ -342,7 +353,15 @@ def _get_specific_subsampling_setting(setting, optional=False):
 rule subsample:
     message:
         """
-        Subsample all sequences into a {wildcards.subsample} set for build '{wildcards.build_name}' with {params.sequences_per_group} per {params.group_by}
+        Subsample all sequences by '{wildcards.subsample}' scheme for build '{wildcards.build_name}' with the following parameters:
+
+         - group by: {params.group_by}
+         - sequences per group: {params.sequences_per_group}
+         - subsample max sequences: {params.subsample_max_sequences}
```

Comment:
so this will be a bit odd because the other parameters will just be the value, but sequences_per_group and subsample_max_sequences will include both the value and the --arg.

Reply:
It's definitely not ideal. Unfortunately, we have a couple other cases where the parameter is actually an argument string for augur filter. I wanted to convey with the message what the possible parameters are and what the defined values are. The trade-off is some redundancy in the output. Here is an example I tried out while testing this branch:

```
        Subsample all sequences by 'another' scheme for build 'global' with the following parameters:

         - group by: country year month
         - sequences per group:
         - subsample max sequences: --subsample-max-sequences 100
         - exclude:
         - include:
         - query: --query "country == 'USA'"
         - priority:
```

Ideally, we'd have implemented our mapping of filter config to arguments more cleanly (maybe like I mocked up in [an earlier comment on this PR](https://github.com/nextstrain/ncov/pull/485#pullrequestreview-472843630) in the section "Wrapping augur commands in the Snakemake config interface").

Maybe a simpler alternative is to define the message like so, such that the argument params only appear when defined:

```
    message:
        """
        Subsample all sequences by '{wildcards.subsample}' scheme for build '{wildcards.build_name}' with the following parameters:

         group by: {params.group_by}
         sequences per group: {params.sequences_per_group}
         subsample max sequences: {params.subsample_max_sequences}
         {params.exclude_argument}
         {params.include_argument}
         {params.query_argument}
         {params.priority_argument}
        """
```

But this might look strange to users, too, when most arguments aren't defined and appear as newlines:

```
        Subsample all sequences by 'another' scheme for build 'global' with the following parameters:

         group by: country year month
         sequences per group:
         subsample max sequences: --subsample-max-sequences 100


         --query "country == 'USA'"

```

Reply:
I think I'd prefer the clarity of the first approach even with the redundancy. Better for users to have double the idea of what's appearing rather than potentially being confused by newlines and 'random' argument values appearing below them, IMO!

Reply:
It seems it would be good to have a mapping of logical config names to 1) parameters for augur, and 2) textual description.